docs/rfcs/0005-json-library.md

```diff
@@ -0,0 +1,48 @@
+---
+layout: page
+title: RFC process
+permalink: /rfc/0005-json-library
+---
+
+* Start date: 2020-07-28
+* RFC PR: TBD
+
+## Summary
+
+Introduce a library for reading and writing JSON into Android Components and Fenix.
+
+## Motivation
+
+We currently interact with JSON in many areas of Android Components and Fenix. Currently serialization and deserialization is all done manually with `JSONObject`, which creates more work for developers and potential bugs. Fenix also is interested in using a library and we'd like to avoid shipping multiple JSON libraries to users. Using a JSON library allows us to reduce code that needs to be written, potentially gain some performance wins from a well maintained lib, and avoid bugs in our custom JSON implementations.
+
+Some examples of JSON usage today are:
+- Reading a `WebAppManifest`
+- Parsing API calls in various service components
+- Sending messages to and from extensions
```

Comment:
One additional motivation for me is that it would be much easier to build streaming JSON parsers. Using `JSONObject` always means loading everything into a String first and then parsing it completely into a `JSONObject` tree. In some areas we have seen memory problems with that approach. There's `JsonReader` in Android, but no one will voluntarily use that. Using a JSON library it is much easier to parse directly from an `InputStream`.